constructor/main.py

```diff
@@ -149,10 +149,16 @@ def main_build(dir_path, output_dir='.', platform=cc_platform,
         if env_name in ("base", "root"):
             raise ValueError(f"Environment name '{env_name}' cannot be used")
         for config_key, value in env_config.copy().items():
-            if isinstance(value, (list, tuple)):
-                env_config[config_key] = [val.strip() for val in value]
             if config_key == "environment_file":
                 env_config[config_key] = abspath(join(dir_path, value))
+            elif config_key == "channels_remap":
+                env_config[config_key] = [
+                    {"src": item["src"].strip(), "dest": "dest".strip()} for item in value
+                ]
+            elif isinstance(value, (list, tuple)):
+                env_config[config_key] = [val.strip() for val in value]
+            else:
+                env_config[config_key] = value
```

Comment:
```suggestion
                env_config[config_key] = value.strip()
```
Should we strip here, too?

Reply:
Not guaranteed to be a `str`. We would need another clause `elif isinstance(...)`.